ports/renesas-ra/machine_pin.c

```diff
@@ -159,29 +161,50 @@ STATIC mp_obj_t machine_pin_obj_init_helper(const machine_pin_obj_t *self, size_
     mp_arg_parse_all(n_args, pos_args, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
 
     // get io mode
-    uint mode = args[0].u_int;
-    if (!IS_GPIO_MODE(mode)) {
-        mp_raise_msg_varg(&mp_type_ValueError, MP_ERROR_TEXT("invalid pin mode: %d"), mode);
+    uint32_t mode;
+    if (args[ARG_mode].u_obj != mp_const_none) {
+        mode = mp_obj_get_int(args[ARG_mode].u_obj);
+        if (!IS_GPIO_MODE(mode)) {
+            mp_raise_msg_varg(&mp_type_ValueError, MP_ERROR_TEXT("invalid pin mode: %d"), mode);
+        }
+    } else {
+        mode = ra_gpio_get_mode(self->pin);
     }
 
     // get pull mode
-    uint pull = 0;
-    if (args[1].u_obj != mp_const_none) {
-        pull = mp_obj_get_int(args[1].u_obj);
+    uint32_t pull;
+    if (args[ARG_pull].u_obj != mp_const_none) {
+        pull = mp_obj_get_int(args[ARG_pull].u_obj);
+        if (!IS_GPIO_MODE(pull)) {
```

Comment:
this should be `IS_GPIO_PULL`

Reply:
Thank you for your review.  I fixed and updated.